packages/near-api-js/src/account.ts

```diff
@@ -579,13 +585,15 @@ export class Account {
             account_id: this.accountId,
             finality: 'optimistic'
         });
+        // Replace raw nonce into a new BN
+        const newResponse = { keys: response?.keys?.map((key) => ({ ...key, access_key: { ...key.access_key, nonce: new BN(key.access_key.nonce) } }))};
         // A breaking API change introduced extra information into the
         // response, so it now returns an object with a `keys` field instead
         // of an array: https://github.com/nearprotocol/nearcore/pull/1789
-        if (Array.isArray(response)) {
-            return response;
+        if (Array.isArray(newResponse)) {
```

Comment:
This check seems irrelevant now that `newResponse` is always an object. Do we need additional logic here to cater for `view_access_key_list` returning either an array or an object, or can we trust that it will always be an object of the shape `{ keys: [] }`